### PR TITLE
chapter_8

### PR DESCRIPTION
**Pull request by muizzuddinyafiq**
ready

### CODE REVIEW

**Comment by muizzuddinyafiq** [comment]
ready

File: app/assets/stylesheets/sessions.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
```
Comment (huongnt-2545) [inline app/assets/stylesheets/sessions.scss]: delete this file, it's not useful

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
```
Comment (huongnt-2545) [inline app/controllers/sessions_controller.rb]:
```suggestion
    if user&.authenticate(params[:session][:password])
```

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  def logged_in?
+    current_user.present?
+  end
+
+  def log_out
+    session.delete(:user_id)
+    @current_user = nil
+  end
+end
+
```
Comment (huongnt-2545) [inline app/helpers/sessions_helper.rb]: remove 2 empty lines

File: app/javascript/stylesheets/application.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload'
```
Comment (huongnt-2545) [inline app/javascript/stylesheets/application.scss]: why you have this file?

File: test/controllers/sessions_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/sessions_controller_test.rb]: delete this file

**Comment by muizzuddinyafiq** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new
+  end
```
Comment (ghost) [inline app/controllers/sessions_controller.rb]:
```suggestion
  def new; end
```

File: config/routes.rb [hunk 7]
```diff
@@ -1,7 +1,11 @@
 Rails.application.routes.draw do
+  get 'sessions/new'
```
Comment (ghost) [inline config/routes.rb]: remove line 2 please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by muizzuddinyafiq** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new; end
+  end
```
Comment (ghost) [inline app/controllers/sessions_controller.rb]: end in this for what?

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ngocvt-0484** [comment]
commented

File: app/controllers/sessions_controller.rb (hunk 8 above)
Comment (ngocvt-0484) [reply app/controllers/sessions_controller.rb]: remove end

**Comment by muizzuddinyafiq** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**